src/ast/mod.rs

```diff
@@ -559,7 +616,7 @@ impl FunctionDecl {
             },
             merge: None,
             merge_action: vec![],
-            default: None,
+            default: Some(Expr::Lit(Literal::Unit)),
```

Comment:
I am curious why the relation now needs a default to work? It does make it a little more cumbersome to define relations with the function syntax. Would you ever want to define a function that returns `Unit` but doesn't have this as a default?

Reply:
Good question- 
`Unit` is a primitive just like `i64`. Before, we had special logic for making default units, making them special in a weird way. Now, I make it explicit that the `relation` syntax is sugar for this function for convenience.

You can still have a function that returns `Unit` and use `set` everywhere.

Reply:
Ah I see.

Is it true then that using a function that does not return a default as an action is a no-op? And that's why you needed the special case before and now with it removed you have added a default?

As a test, this rule seems to have no effect, when there is no default (based on the viz):

```
(datatype Expr)

(function trigger () Expr)

(function foo () i64)

(trigger)

(rule ((= x (trigger))) ((foo)))

(run 1)
```

But it does when there is a default:


```
(datatype Expr)

(function trigger () Expr)

(function foo () i64 : default 10)

(trigger)

(rule ((= x (trigger))) ((foo)))

(run 1)
```

Reply:
Hmm, shouldn't you get an exception when there isn't a default?
Otherwise, the action might fail halfway-through!

Reply:
Yeah, so if I am understanding that correctly, only functions with defaults can be used as actions?